src/util/router.js

```diff
@@ -87,7 +130,8 @@ export const forceReplace = ({ router, unsavedChanges }, location) => {
 // Returns the resources that would be preserved after navigating from the
 // `from` route to the `to` route.
 export const preservedData = (to, from, requestData) => {
-  if (from === START_LOCATION) return requestData.resources;
+  if (from === START_LOCATION || to.path === from.path)
```

Comment:
This change is tested in test/router.spec.js. It wasn't needed before despite our use of filter query parameters because we preserve data after navigation between tabs — even if the navigation is from the same tab to itself.